diff --git a/src/torchcodec/_core/Encoder.cpp b/src/torchcodec/_core/Encoder.cpp
index 2fe136fb..d024834b 100644
--- a/src/torchcodec/_core/Encoder.cpp
+++ b/src/torchcodec/_core/Encoder.cpp
@@ -846,7 +846,8 @@ void VideoEncoder::initializeEncoder(
   if (videoStreamOptions.pixelFormat.has_value()) {
     // TODO-VideoEncoder: (P2) Enable pixel formats to be set by user on GPU
     // and handled with the appropriate NPP function on GPU.
-    if (frames_.device().type() == kStableCUDA) {
+    if (frames_.device().type() == kStableCUDA ||
+        frames_.device().type() == kStableXPU) {
       STD_TORCH_CHECK(
           false,
           "Video encoding on GPU currently only supports the nv12 pixel format. "
@@ -855,7 +856,8 @@ void VideoEncoder::initializeEncoder(
     outPixelFormat =
         validatePixelFormat(*avCodec, videoStreamOptions.pixelFormat.value());
   } else {
-    if (frames_.device().type() == kStableCUDA) {
+    if (frames_.device().type() == kStableCUDA ||
+        frames_.device().type() == kStableXPU) {
       // Default to nv12 pixel format when encoding on GPU.
       outPixelFormat = DeviceInterface::CUDA_ENCODING_PIXEL_FORMAT;
     } else {
@@ -910,7 +912,8 @@ void VideoEncoder::initializeEncoder(
         0);
   }
 
-  if (frames_.device().type() == kStableCUDA) {
+  if (frames_.device().type() == kStableCUDA ||
+      frames_.device().type() == kStableXPU) {
     deviceInterface_->registerHardwareDeviceWithCodec(avCodecContext_.get());
     deviceInterface_->setupHardwareFrameContextForEncoding(
         avCodecContext_.get());
@@ -1208,7 +1211,7 @@ void MultiStreamEncoder::initializeVideoStream() {
   if (videoStream.options.pixelFormat.has_value()) {
     // TODO-MultiStreamEncoder: (P2) Enable pixel formats to be set by user on
     // GPU and handled with the appropriate NPP function on GPU.
-    if (deviceType == kStableCUDA) {
+    if (deviceType == kStableCUDA || deviceType == kStableXPU) {
       STD_TORCH_CHECK(
           false,
           "Video encoding on GPU currently only supports the nv12 pixel format. "
@@ -1217,7 +1220,7 @@ void MultiStreamEncoder::initializeVideoStream() {
     outPixelFormat =
         validatePixelFormat(*avCodec, videoStream.options.pixelFormat.value());
   } else {
-    if (deviceType == kStableCUDA) {
+    if (deviceType == kStableCUDA || deviceType == kStableXPU) {
       // Default to nv12 pixel format when encoding on GPU.
       outPixelFormat = DeviceInterface::CUDA_ENCODING_PIXEL_FORMAT;
     } else {
@@ -1275,7 +1278,7 @@ void MultiStreamEncoder::initializeVideoStream() {
         0);
   }
 
-  if (deviceType == kStableCUDA) {
+  if (deviceType == kStableCUDA || deviceType == kStableXPU) {
     videoStream.deviceInterface->registerHardwareDeviceWithCodec(
         videoStream.avCodecContext.get());
     videoStream.deviceInterface->setupHardwareFrameContextForEncoding(
